src/storage/mod.rs

```diff
@@ -1016,9 +1016,15 @@ mod tests {
         })
     }
 
-    fn expect_fail<T>(done: Sender<i32>, id: i32) -> Callback<T> {
+    fn expect_fail<T>(done: Sender<i32>, id: i32, expect_err: &'static str) -> Callback<T> {
         Box::new(move |x: Result<T>| {
-            assert!(x.is_err());
+            match x {
+                Err(e) => {
+                    let err = format!("{:?}", e);
+                    assert_eq!(&err, expect_err);
```

Comment:
maybe we can use Failure to simply this, I think it is bad to check the error message directly.